From c9af695202cde316d3e9ab1fa6d295a967a1efb3 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Thu, 3 Dec 2015 11:46:46 +0530
Subject: [PATCH 01/12] SQUASHME! [VPG]: drm/i915: Fix KW issue on encoder's
 NULL ptr check

KW static analyzer warns of passing 'encoder' ptr to get_power_domain()
without checking for NULL. This patch fixes it by moving the
get_power_domain() call after the encoder's NULL check.

This patch should be merged to the following patch
SQUASHME! - This patch should be squashed into the following existing patch:
Author:    Thulasimani,Sivakumar <sivakumar.thulasimani@intel.com>
Date:      Tue, 15 Sep 2015 20:05:05 +0530
Change-Id: b7072760d789cd71c9922fd119fae1a90a15fee0
Subject:   SQUASHME [VPG]: drm/i915: Dis/Enable Phy in upfront linktraining

Change-Id: I5d41208c2a010b9e0f4f201452555fe91ed63c04
Signed-off-by: Thulasimani,Sivakumar <sivakumar.thulasimani@intel.com>
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-10950
Reviewed-on: https://android.intel.com:443/444118
---
 drivers/gpu/drm/i915/intel_display.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d5b163a..829028a 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -15211,13 +15211,14 @@ bool chv_upfront_link_train(struct drm_device *dev,
 	bool found = false;
 	bool valid_crtc = false;
 	uint8_t tmp_lane_count, tmp_link_bw;
-	power_domain = intel_display_port_power_domain(encoder);
 
 	if (!connector || !encoder) {
 		DRM_DEBUG_KMS("dp connector/encoder is NULL\n");
 		return false;
 	}
 
+	power_domain = intel_display_port_power_domain(encoder);
+
 	/* If we already have a crtc, start link training directly */
 	if (crtc) {
 		valid_crtc = true;
-- 
1.9.1

